### PR TITLE
feat: Junit parser for Screwdriver

### PR DESCRIPTION
**Pull request by rchopde**
## Context

Open Source Screwdriver is the flagship CI/CD platform from Verizon Media. This source code publishes a shared command for Screwdriver which provides key functionality for parsing test results. This functionality provided by my source code aligns with Screwdriver's philosophy of using shared commands to extend it's capability.
This also enables open source users of Screwdriver an easier migration path from tools like Jenkins where similar functionality is provided by it's plugin infrastructure. This also aligns with Screwdriver product goal of eventually building a market place for Shared Commands.

## Objective

This source code publishes a shared command for Screwdriver which provides key functionality for parsing test results. This functionality provided by my source code aligns with Screwdriver's philosophy of using shared commands to extend it's capability.

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by tkyi** [comment]
There seems to be 2 PRs with the same file names. Should we close the first one?

**Comment by tkyi** [comment]
I think it would be good to add instructions in the README for how to use this code in a command

File: process-surefire-report.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#!/bin/sh
+  
+set -e
+
+if [ $# -ne 2 ] || ! [ -d "$1" ]; then
+  echo "Usage: $0 <path-to-surefire-reports> <update-sd-ui [true | false]>" >&2
+  exit -1
+fi
+
+if ! [ -f "/usr/bin/ruby" ]; then
```
Comment (jithine) [inline process-surefire-report.sh]: please switch the check to `command -v ruby >/dev/null 2>&1`

https://stackoverflow.com/questions/592620/how-to-check-if-a-program-exists-from-a-bash-script

File: process-surefire-report.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  echo "Installing ruby rpm..."
+  sudo yum install --debuglevel=1 -y ruby
```
Comment (jithine) [inline process-surefire-report.sh]: we can start with this, later on can expand to other package manager checks like `apt-get`

**Review by jithine — COMMENTED** [review]
Also please add a screwdriver.yaml for publishing this. 

See https://github.com/screwdriver-cd-test/functional-commands/blob/master/screwdriver.yaml for example

**Review by jithine — APPROVED** [review]

**Review by jithine — APPROVED** [review]

**Review by tkyi — APPROVED** [review]

File: sd-command.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+name: junit-tests
+version: '1.0'
+description: |
+  Process junit reports in surefire or touchstone format to compute total number of pass, failed tests and optionally updated SDV4 UI
+usage: |
+  sd-cmd exec screwdriver-cd/junit-tests@<VERSION|TAG> <JUNIT REPORTS DIRECTORY> <UPDATE SD UI [TRUE|FALSE] >
+  Args:
+          DIRECTORY             Directory to surefire reports [surefire | touchstone].
+          SDUI                  Update SDV4 UI in a format fail/total tests.
+maintainer: rchopde@verizonmedia.com
```
Comment (tkyi) [inline sd-command.yaml:11]: should this be a non-work email?

Comment (rchopde) [reply sd-command.yaml:11]: I added  rchopde@verizonmedia.com to my account in addition to my personal.

Comment (jithine) [reply sd-command.yaml:11]: As long that github user has this email in their account, should be okay.

**Pull request merged**